### ISSUE TITLE
Is feature selection step necessary before XGBoost?

### ISSUE THREAD

**Issue by yilinwu123** [dmlc/xgboost#7718]
Dear Developer:

I am interested in using 'xgboost' package to do classification on high dimensional gene expression data. May I ask whether it is helpful to do additional feature seleciton steps before using xgboost since xgboost algorithm can also select important features? 

I tried a feature selection method called MRMR (Maximum Relevance — Minimum Redundancy) to remove noisy and redundant features before using xgboost. However, I got a lower classification accuracy when using feature selection method 'MRMR' than the results without using 'MRMR'. Does this mean this additional feature selection step is not helpful and I don't need to use feature selection before doing classificaiton with 'xgboost'? 

Thank you so much for your help!

**Comment by Craigacp** [comment]
The mRMR algorithm can't find features which have positive interactions (i.e. ones which provide more information jointly than they do separately). XGBoost as it is based on decision trees can exploit this kind of feature interaction, and so using mRMR first may remove features XGBoost finds useful. There are other information theoretic feature selection algorithms which don't have this issue, but in general I'd probably not bother with feature selection before running XGBoost, and instead tune the regularisation and tree depth parameters of XGBoost to achieve a smaller feature set.

**Comment by yilinwu123** [comment]
> The mRMR algorithm can't find features which have positive interactions (i.e. ones which provide more information jointly than they do separately). XGBoost as it is based on decision trees can exploit this kind of feature interaction, and so using mRMR first may remove features XGBoost finds useful. There are other information theoretic feature selection algorithms which don't have this issue, but in general I'd probably not bother with feature selection before running XGBoost, and instead tune the regularisation and tree depth parameters of XGBoost to achieve a smaller feature set.

Dear Adam:

Thanks a lot for your reply. Your suggestions are very helpful. 

So for high dimensional data with small sample size (e.g. 200 samples with 3000 features), is it okay to skip feature selection steps and do classification directly? I tried to focus on tuning the regularisation and tree depth parameters, it actually performed better than adding feature selection step, although there seemed to be some overfitting problems.

If I may ask, do information theoretic feature selection algorithms use some measure to assess the feature interactions (e.g. mutual information)? 

Thanks again for your help! I really appreciate it!

**Comment by Craigacp** [comment]
I think with many more features than examples most things will overfit a bit as there are too many ways of making spurious correlations. Properly regularised models will help, as can feature selection, but I wouldn't recommend mRMR if you want to use tree ensembles to make the final prediction.

Yes, information theoretic feature selection algorithms use entropies or mutual informations to measure the feature interactions. I wrote a journal paper surveying the different algorithms about 10 years ago during my PhD if you want to read more about them - https://www.jmlr.org/papers/volume13/brown12a/brown12a.pdf.

**Comment by yilinwu123** [comment]
> I think with many more features than examples most things will overfit a bit as there are too many ways of making spurious correlations. Properly regularised models will help, as can feature selection, but I wouldn't recommend mRMR if you want to use tree ensembles to make the final prediction.
> 
> Yes, information theoretic feature selection algorithms use entropies or mutual informations to measure the feature interactions. I wrote a journal paper surveying the different algorithms about 10 years ago during my PhD if you want to read more about them - https://www.jmlr.org/papers/volume13/brown12a/brown12a.pdf.

Dear Adam:

It is very helpful. I will read this paper. Thank you so much for your suggestions. I really appreciate it!

**Comment by trivialfis** [comment]
Thank you for the interesting discussion!  I really enjoy the paper. ;-)
